crates/nix-ninja/src/task.rs

```diff
@@ -587,12 +578,19 @@ fn extract_store_paths(store_regex: &Regex, s: &str) -> Result<Vec<StorePath>> {
     Ok(store_paths)
 }
 
-fn new_opaque_file(nix: &NixTool, path: PathBuf) -> Result<DerivedFile> {
+fn new_opaque_file(nix: &NixTool, build_dir: &PathBuf, path: PathBuf) -> Result<DerivedFile> {
+    let relative_path = match relative_from(&path, build_dir) {
```

Comment:
Small style nit: instead of `match` you can use `unwrap_or(path)`
```suggestion
    let relative_path = relative_from(&path, build_dir).unwrap_or(path);
```

Reply:
Follow up in: https://github.com/pdtpartners/nix-ninja/pull/28